fury/actors/odf_slicer.py

```diff
@@ -0,0 +1,257 @@
+        """
+        Set visible volume from x1 (inclusive) to x2 (inclusive),
+        y1 (inclusive) to y2 (inclusive), z1 (inclusive) to z2
+        (inclusive).
+        """
+        mask = np.zeros(self.grid_shape, dtype=bool)
+        mask[x1:x2 + 1, y1:y2 + 1, z1:z2 + 1] = True
+        self.mask = mask
+
+        self._update_mapper()
+
+    def slice_along_axis(self, slice_index, axis='zaxis'):
+        """
+        Slice ODF field at given `slice_index` along axis
+        in ['xaxis', 'yaxis', zaxis'].
+        """
+        if axis == 'xaxis':
+            self.display_extent(slice_index, slice_index,
+                                0, self.grid_shape[1] - 1,
+                                0, self.grid_shape[2] - 1)
+        elif axis == 'yaxis':
+            self.display_extent(0, self.grid_shape[0] - 1,
+                                slice_index, slice_index,
+                                0, self.grid_shape[2] - 1)
+        elif axis == 'zaxis':
+            self.display_extent(0, self.grid_shape[0] - 1,
+                                0, self.grid_shape[1] - 1,
+                                slice_index, slice_index)
+        else:
+            raise ValueError('Invalid axis name {0}.'.format(axis))
+
+    def display(self, x=None, y=None, z=None):
+        """
+        Display a slice along x, y, or z axis.
+        """
+        if x is None and y is None and z is None:
+            self.slice_along_axis(self.grid_shape[2]//2)
+        elif x is not None:
+            self.slice_along_axis(x, 'xaxis')
+        elif y is not None:
+            self.slice_along_axis(y, 'yaxis')
+        elif z is not None:
+            self.slice_along_axis(z, 'zaxis')
+
+    def update_sphere(self, vertices, faces, B):
+        """
+        Dynamically change the sphere used for SH to SF projection.
+        """
+        if self.B is None:
+            raise ValueError('Can\'t update sphere when using '
+                             'SF coefficients.')
+        self.vertices = vertices
+        if self.affine is not None:
+            self.w_verts = self.vertices.dot(self.affine[:3, :3])
+        self.faces = faces
+        self.B = B
+
+        # draw ODFs with new sphere
+        self._update_mapper()
+
+    def _update_mapper(self):
+        """
+        Map vtkPolyData to the actor.
+        """
+        polydata = vtk.vtkPolyData()
+
+        offsets = self._get_odf_offsets(self.mask)
+        if len(offsets) == 0:
+            self.mapper.SetInputData(polydata)
+            return None
+
+        sph_dirs = self._get_sphere_directions()
+        sf = self._get_sf(self.mask)
+
+        all_vertices = self._get_all_vertices(offsets, sph_dirs, sf)
+        all_faces = self._get_all_faces(len(offsets), len(sph_dirs))
+        all_colors = self._generate_color_for_vertices(sf)
+
+        set_polydata_triangles(polydata, all_faces)
+        set_polydata_vertices(polydata, all_vertices)
+        set_polydata_colors(polydata, all_colors)
```

Comment:
there is a lot of deep copy here. I think this part can be easily optimized. There are some examples in the `network` examples.

For now, it is fine, but can you add a small note/TODO. It deserves its own PR